ui-v2/src/hooks/global-concurrency-limits.test.tsx

```diff
@@ -112,4 +117,202 @@ describe("global concurrency limits hooks", () => {
+		const queryClient = new QueryClient();
+		const MOCK_NEW_LIMIT_ID = "1";
+		const MOCK_NEW_LIMIT = {
+			active: true,
+			active_slots: 0,
+			denied_slots: 0,
+			limit: 0,
+			name: "global concurrency limit 1",
+			slot_decay_per_second: 0,
+		};
+
+		// ------------ Mock API requests after queries are invalidated
+		const NEW_LIMIT_DATA = {
+			...MOCK_NEW_LIMIT,
+			id: MOCK_NEW_LIMIT_ID,
+			created: "2021-01-01T00:00:00Z",
+			updated: "2021-01-01T00:00:00Z",
+			active_slots: 0,
+			slot_decay_per_second: 0,
+		};
+
+		const mockData = [
+			...seedGlobalConcurrencyLimits(),
+			{
+				...MOCK_NEW_LIMIT,
+				...NEW_LIMIT_DATA,
+			},
+		];
+		mockFetchGlobalConcurrencyLimitsAPI(mockData);
+
+		// ------------ Initialize cache
+		queryClient.setQueryData(
+			buildListGlobalConcurrencyLimitsQuery(filter).queryKey,
+			seedGlobalConcurrencyLimits(),
+		);
+
+		// ------------ Initialize hooks to test
+		const { result: useListGlobalConcurrencyLimitsResult } = renderHook(
+			() => useListGlobalConcurrencyLimits(filter),
+			{ wrapper: createQueryWrapper({ queryClient }) },
+		);
+		const { result: useCreateGlobalConcurrencyLimitResult } = renderHook(
+			useCreateGlobalConcurrencyLimit,
+			{ wrapper: createQueryWrapper({ queryClient }) },
+		);
+
+		// ------------ Invoke mutation
+		act(() =>
+			useCreateGlobalConcurrencyLimitResult.current.createGlobalConcurrencyLimit(
+				MOCK_NEW_LIMIT,
+			),
+		);
+
+		// ------------ Assert
+		await waitFor(() =>
+			expect(useCreateGlobalConcurrencyLimitResult.current.isSuccess).toBe(
+				true,
+			),
+		);
+		expect(useListGlobalConcurrencyLimitsResult.current.data).toHaveLength(2);
+		const newLimit = useListGlobalConcurrencyLimitsResult.current.data?.find(
+			(limit) => limit.id === MOCK_NEW_LIMIT_ID,
+		);
+		expect(newLimit).toMatchObject(NEW_LIMIT_DATA);
+	});
+
+	/**
+	 * Data Management:
+	 * - Asserts update mutation API is called.
+	 * - Upon update mutation API being called, cache invalidates global concurrency limit details cache
+	 */
+	it("useUpdateGlobalConcurrencyLimit() invalidates cache and fetches updated value", async () => {
+		const queryClient = new QueryClient();
+		const MOCK_UPDATE_LIMIT_ID = "0";
+		const UPDATED_LIMIT_BODY = {
+			active: true,
+			active_slots: 0,
+			denied_slots: 0,
+			limit: 0,
+			name: "global concurrency limit updated",
+			slot_decay_per_second: 0,
+		};
+		const UPDATED_LIMIT = {
+			...UPDATED_LIMIT_BODY,
+			id: MOCK_UPDATE_LIMIT_ID,
+		};
+
+		// ------------ Mock API requests after queries are invalidated
+		const mockData = seedGlobalConcurrencyLimits().map((limit) =>
+			limit.id === MOCK_UPDATE_LIMIT_ID ? UPDATED_LIMIT : limit,
+		);
+		mockFetchGlobalConcurrencyLimitsAPI(mockData);
+		mockFetchGlobalConcurrencyLimitDetailsAPI(UPDATED_LIMIT);
+
+		// ------------ Initialize cache
+		queryClient.setQueryData(
+			buildGetGlobalConcurrencyLimitQuery(MOCK_UPDATE_LIMIT_ID).queryKey,
```

Comment:
If we're only importing `buildGetGlobalConcurrencyLimitQuery` to get the query key, would it make sense to import the query key factory instead? I expect that we'll need to share query keys between modules in some cases.

Reply:
I think either works.

Maybe for this case, using the query key factor makes more sense.

The only benefit for doing it this way is if we were to do `queryClient.setQueryData()` or `queryClient.get()`, there's some typescript magic that knows what the expected type is with the `queryOptions` object.

>  I expect that we'll need to share query keys between modules in some cases.
I feel like this is something we may want to avoid. I think we should design our queryKeys to only group entities that are dependent with each other. But that could simply just mean placing all the dependent queries and mutations together in a folder / module

Reply:
Updated in latest push to use the `queryKeyFactory` instead